zk-token-sdk/src/instruction/mod.rs

```diff
@@ -51,7 +50,7 @@ pub trait ZkProofData: Pod {
 }
 
 pub trait ZkProofContext: Pod {
-    fn proof_type(&self) -> Result<ProofType, InstructionError>;
+    const PROOF_TYPE: ProofType;
```

Comment:
Yes this is a good point. I was initially a little torn with this because there should technically be one `ZkProofContext` type for each `ZkProofData` type. Even if `ZkProofContext` is an empty market trait, having it as an associated type in `ZkProofData` naturally enforces this. For example, implementing `CloseProofContext` inside `TransferProofData` would not really make sense.

But in the end, I decided to go with removing `ZkProofContext` because it does simplify the instruction implementations a bit. Thanks for the suggestion!